File: ocean2city-backend/SubQuip.Model/Order/OrderDetailViewModel.cs
using Ocean2City.ViewModel.CartItem;
using Ocean2City.ViewModel.User;
using System;
using System.Collections.Generic;
using System.Text;

namespace Ocean2City.ViewModel.Order
{
    public class OrderDetailViewModel
    {
        public string OrderId { get; set; }

        public string UserId { get; set; }

        public string AddressId { get; set; }

        public DateTime DeliveryDate { get; set; }

        public string DeliveryTiming { get; set; }
        
        public bool IsDelivered { get; set; }
        
        public double TotalAmount { get; set; }

        public List<CartItemViewModel> cartItems { get; set; }
    }

    public class Order
    {
        public UserViewModel UserViewModel { get; set; }

        public UserAddressViewModel UserAddressViewModel { get; set; }

        public List<CartItemViewModel> CartItemList { get; set; }

        public OrderDetailViewModel OrderDetailViewModel { get; set; }
    }
}

File: ocean2city-frontend/src/app/webapi/models/order-detail-view-model.ts
/* tslint:disable */
import { CartItemViewModel } from './cart-item-view-model';
export interface OrderDetailViewModel {
  orderId?: string;
  userId?: string;
  addressId?: string;
  deliveryDate?: string;
  deliveryTiming?: string;
  isDelivered?: boolean;
  totalAmount?: number;
  cartItems?: Array<CartItemViewModel>;
}
